src/StackExchange.Redis/RedisDatabase.cs

```diff
@@ -3116,32 +3152,67 @@ private Message GetSortedSetCombineAndStoreCommandMessage(SetOperation operation
 
-            List<RedisValue>? values = null;
+            List<RedisValue>? values = new List<RedisValue>();
+            AddWeightsAndAggregation(ref values, weights, aggregate);
+            return new SortedSetCombineAndStoreCommandMessage(Database, flags, command, destination, keys, values?.ToArray() ?? RedisValue.EmptyArray);
+        }
+
+        private Message GetSortedSetCombineCommandMessage(SetOperation operation, RedisKey[] keys, double[]? weights, Aggregate aggregate, bool withScores, CommandFlags flags)
+        {
+            var command = operation switch
+            {
+                SetOperation.Intersect => RedisCommand.ZINTER,
+                SetOperation.Union => RedisCommand.ZUNION,
+                SetOperation.Difference => RedisCommand.ZDIFF,
+                _ => throw new ArgumentOutOfRangeException(nameof(operation)),
+            };
+            if (keys == null) throw new ArgumentNullException(nameof(keys));
+            // Difference operation can't have weights or aggregation
+            if (operation == SetOperation.Difference && (weights != null || aggregate != Aggregate.Sum))
+                throw new ArgumentException("ZDIFF cannot be used with weights or with aggregation.");
+
+            List<RedisValue>? values = new List<RedisValue>();
+            values.Add(keys.Length);
+            foreach(var key in keys)
+                values.Add(key.AsRedisValue());
+            AddWeightsAndAggregation(ref values, weights, aggregate);
+            if (withScores)
+            {
+                values.Add(RedisLiterals.WITHSCORES);
+            }
+            return Message.Create(Database, flags, command, values?.ToArray() ?? RedisValue.EmptyArray);
+        }
+
+        private void AddWeightsAndAggregation(ref List<RedisValue> args, double[]? weights, Aggregate aggregate)
+        {
             if (weights != null && weights.Length != 0)
             {
-                (values ??= new List<RedisValue>()).Add(RedisLiterals.WEIGHTS);
+                (args ??= new List<RedisValue>()).Add(RedisLiterals.WEIGHTS);
                 foreach (var weight in weights)
-                    values.Add(weight);
+                    args.Add(weight);
             }
             switch (aggregate)
             {
                 case Aggregate.Sum: break; // default
                 case Aggregate.Min:
-                    (values ??= new List<RedisValue>()).Add(RedisLiterals.AGGREGATE);
-                    values.Add(RedisLiterals.MIN);
+                    (args ??= new List<RedisValue>()).Add(RedisLiterals.AGGREGATE);
+                    args.Add(RedisLiterals.MIN);
                     break;
                 case Aggregate.Max:
-                    (values ??= new List<RedisValue>()).Add(RedisLiterals.AGGREGATE);
-                    values.Add(RedisLiterals.MAX);
+                    (args ??= new List<RedisValue>()).Add(RedisLiterals.AGGREGATE);
+                    args.Add(RedisLiterals.MAX);
```

Comment:
Let's leave this as `values` please - less diff and the meaning is "RedisValue" by the word so was correct.